app/models/file_set.rb

```diff
@@ -4,4 +4,8 @@ class FileSet < Valhalla::Resource
   attribute :id, Valkyrie::Types::ID.optional
   attribute :title, Valkyrie::Types::Set
   attribute :member_ids, Valkyrie::Types::Array
+
+  def thumbnail_id
+    @thumbnail_id ||= Valkyrie.config.metadata_adapter.query_service.find_members(resource: self).find { |x| x.use.include?(Valkyrie::Vocab::PCDMUse.ServiceFile) }.try(:id)
```

Comment:
I want to have it not return the whole URL so we have only one place generating manifest URLs, which isn't attached to the model itself, so that it can vary between dev/staging. However, I moved some code around so hopefully the intent is more clear - have a look/